File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.cpp
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        unordered_map<int,int> mp;
        for(int i:arr){
            if(mp.find(i) != mp.end())
                mp[i]++;
            else
                mp[i]=1;
        }
        int missing=0;
        while(k){
            missing++;
            if(mp.find(missing) == mp.end())
            {k--;}
        }
        return missing;
    }
};

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.java
class Solution {
    public int findKthPositive(int[] arr, int k) {
        int start=0,end=arr.length,n=arr.length;
        int diff=0,mid=0;
        while(start < end){
            mid=(start+end)/2;
            diff = arr[mid]-mid-1;
            if(diff < k){
                start=mid+1;
            }else{
                end=mid;
            }
        }
        return start+k;
    }
}